misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java

```diff
@@ -440,18 +465,41 @@ private void execute(boolean buildWithTests) {
             // Call module stop and test stop function
             suite.getInitFunction().invokeStopFunctions();
         });
+
+        // Coverage report handling
+        if (!buildWithTests && coverageFlag) {
+
+            try {
+
+                CoverageManager coverageManager = CoverageManager.getInstance();
+                CoverageDataFormatter coverageDataFormatter = coverageManager.getCoverageDataFormatter();
+                List<LCovData> packageLCovDataList = coverageDataFormatter.getFormattedCoverageData(
+                        coverageManager.getExecutedInstructionOrderMap(), testSuites);
+                coverageDataFormatter.writeFormattedCovDataToFile(packageLCovDataList, sourceRoot);
+
+            } catch (Throwable e) {
+                String errorMsg = String.format("\t[fail] [coverage report generation] :\t    " +
+                        "%s", TesterinaUtils.formatError(e.getMessage()));
+                errStream.println(errorMsg);
+            }
+
```

Comment:
Shall we remove all additional empty lines here?

Reply:
Some are removed not all